odoo/addons/base/models/ir_rule.py

```diff
@@ -159,6 +164,12 @@ def _compute_domain_context_values(self):
                 v = tuple(v)
             yield v
 
+    def _validate_domain(self, domain, eval_context={}):
+        try:
+            return expression.normalize_domain(safe_eval(domain, eval_context) if domain else [])
```

Comment:
It doesn't check the validity of the domain for the targeted model. If someone write a domain with a invalid field, the crash will occur anyway (with a different error but still). It is acceptable ?

Maybe we should call `_search` (shouldn't make any SQL queries most of the time) of the targeted model to ensure that the domain won't crash later. In this case, you should split the code `_check_domain_force` and `_compute_domain` (IMO, you don't need to check the validity of the domain during `_compute_domain`)

Reply:
@sash-odoo